core/ip/mac.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ip
+
+import (
+	"net"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// GetMACAddressForIP finds MAC address
+func GetMACAddressForIP(ip string) (string, error) {
+	var currentNetworkHardwareName string
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to retrieve network interfaces")
+	}
+
+	for _, i := range interfaces {
+		if addresses, err := i.Addrs(); err == nil {
+			for _, addr := range addresses {
+				// only interested in the name with current IP address
+				if strings.Contains(addr.String(), ip) {
+					currentNetworkHardwareName = i.Name
```

Comment:
As soon as we get a matching IP we should terminate the loop. There cannot be more than one interface with the same IP address on a system.

Reply:
Also would suggest naming:
```suggestion
					matchedInterface = i.Name
```